app/controllers/api/automate_domains_controller.rb

```diff
@@ -1,5 +1,21 @@
 module Api
   class AutomateDomainsController < BaseController
+    def delete_resource(type, id = nil, _data = {})
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        domain = resource_search(id, type, klass)
+        api_log_info("Deleting #{automate_domain_ident(domain)}")
+
+        begin
+          MiqAeDomain.where(:name => domain.name).each(&:destroy)
```

Comment:
@jvlcek I think we have something called readonly domains like ManageIQ and  Redhat should the user be allowed to delete these domains. Or should this be allowed only for user domains?

Reply:
@jvlcek is this going to use some RBAC check to see if a user is allowed to delete a domain.

Reply:
@jvlcek I think instead of destroy it might be better to call destroy_queue since we would want to delete the git directories if the domain is a git based domain.
https://github.com/ManageIQ/manageiq-automation_engine/blob/e136b74e9b2005d050149d9c7019784a531d82e1/app/models/miq_ae_domain.rb#L162

Reply:
Thank you for your help @mkanoor! It is very much appreciated.
I have pushed a change to address two of your three comments.
I will push the resolution the the RBAC check issue and an updated spec shortly.

JoeV

Reply:
@mkanoor RBAC is already being handled by the change to the `config/api.yml` file.
e.g.:
https://github.com/ManageIQ/manageiq-api/pull/548/files#diff-15502b585d5033aa8043cc26237a06baR257

`:identifier: miq_ae_domain_delete`

So the user must be associated with a role that has the `miq_ae_domain_delete` product feature enabled.

So RBAC `is` being check. 

Thank you.